seismicpro/containers.py

```diff
@@ -274,6 +275,151 @@ def apply(self, func, cols, res_cols=None, axis=None, unpack_args=False, inplace
+        format : "fwf" or "csv", optional, defaults to "fwf"
+            Format of the file with headers. Currently, the following options are supported:
+            * "fwf" - fixed-width format,
+            * "csv" - comma-separated values format.
+        has_header : bool, optional, defaults to False
+            Indicate if the first row of the file contains header names or not.
+        usecols : array-like of int or None, optional, defaults to None
+            Columns indices to be selected from the file. Unlike `pandas` loaders, it is allowed to use negative
+            indices. Should be always passed in ascending order and have the same length as `headers` if both passed.
+        sep : str, defaults to ','
+            Separator used in the file. Used only for "csv" `format`.
+       skiprows : int, optional, defaults to 0
+            Number of rows to skip from the beginning of the file.
+        decimal : str, optional, defaults to None
+            Decimal point character. If not provided, it will be inferred from the file. Used only for "fwf" `format`.
+        encoding : str, optional, defaults to "UTF-8"
+            File encoding.
+        keep_all_headers : bool, optional, defaults to False
+            Whether to keep headers for traces that were missed in the loaded file.
+        inplace : bool, optional, defaults to False
+            Whether to load headers inplace or to a copy.
+        **kwargs : misc, optional
+            Additional arguments for loading function. If `format="fwf"`, passed to `pandas.read_csv`.
+            If `format="csv"`, passed to `polars.read_csv`.
+
+        Returns
+        -------
+        result : same type as self
+            `self` with the loaded headers.
+
+        Raises:
+        -------
+        ValueError
+            If the `format` argument is not one of the supported formats ('fwf', 'csv').
+        """
+        if format not in ["fwf", "csv"]:
+            raise ValueError(f"Unknown format `{format}`, available formats are ('fwf', 'csv')")
+
+        self = maybe_copy(self, inplace, ignore="headers")  # pylint: disable=self-cls-assignment
+
+        if usecols is not None or decimal is None:
+            with open(path, 'r', encoding=encoding) as f:
+                n_skip = 1 + has_header + skiprows
+                row = [next(f) for _ in range(n_skip)][-1]
+            # If decimal is not provided, try inferring it from the file
+            if decimal is None:
+                decimal = '.' if '.' in row else ','
+            if usecols is not None:
+                usecols = np.asarray(usecols)
+                if any(usecols < 0):  # Processing negative `usecols`
+                    usecols_sep = sep if format == "csv" else None
+                    # Find the position of columns with negative `usecols` by adding them to the total number of
+                    # columns in the file.
+                    usecols[usecols < 0] += len(row.split(usecols_sep))
```

Comment:
Wouldn't this be simpler?
```suggestion
                    usecols = np.arange(len(row.split(usecols_sep)))[usecols]
```

Reply:
added